src/dishka/integrations/aiogram.py

```diff
@@ -38,14 +39,44 @@ def __init__(self, container):
         self.container = container
 
     async def __call__(
-            self, handler, event, data,
+        self, handler, event, data,
     ):
         async with self.container({TelegramObject: event}) as sub_container:
             data[CONTAINER_NAME] = sub_container
             return await handler(event, data)
 
 
-def setup_dishka(container: AsyncContainer, router: Router) -> None:
+class AutoInjectMiddleware(BaseMiddleware):
+    @staticmethod
+    def _analysis_params(func) -> set[str]:
+        new_params = [CONTAINER_NAME]
+        for key, type_hint in func.__annotations__.items():
```

Comment:
looks like we here are some problems. At least you remove any Annotated insetad of those which are provessed by dishka. 

Let's instead take new function and pass it to `HandlerObject`. And then we can try to copy params and varkw to old handler.